### PR TITLE
Added UI

### PR DESCRIPTION
**Pull request by mfborman**
Made UI which can take in user input character by character, and interpret the string, deciding what action the user wants to take.
TODO:
Currently DOES NOT scan. 
Currently DOES NOT implement oi_t in action functions

### CODE REVIEW

**Comment by zwbce** [comment]
+1

**Comment by jwfehr** [comment]
+1

File: UI.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+#include "bluetooth.h"
+#include "movement.h"
+
+/// Read user input character by character and store in char array
+/// user_input_string
+void
+read_raw_UI()
+{
+    int cur_char;
+    int i;
```
Comment (ghost) [inline UI.c:20]: init to 0

Comment (jwfehr) [inline UI.c:20]: Pull Request Closed this Error Needs to be fixed first

**Pull request closed without merging**